include/profiler.h

```diff
@@ -142,20 +142,103 @@ void profiler_log_start(struct log_event_buf *buf);
 static inline void profiler_log_start(struct log_event_buf *buf) {}
 #endif
 
+/** @brief Encode and add uint32_t data type to a buffer.
+ *
+ * @warning The buffer must be initialized with @ref profiler_log_start
+ *          before calling this function.
+ *
+ * @param buf Pointer to the data buffer.
+ * @param data Data to add to the buffer.
+ */
+#ifdef CONFIG_PROFILER
+void profiler_log_encode_uint32(struct log_event_buf *buf, const uint32_t data);
+#else
+static inline void profiler_log_encode_uint32(struct log_event_buf *buf,
+					      const uint32_t data) {}
+#endif
+
+/** @brief Encode and add int32_t data type to a buffer.
+ *
+ * @warning The buffer must be initialized with @ref profiler_log_start
+ *          before calling this function.
+ *
+ * @param buf Pointer to the data buffer.
+ * @param data Data to add to the buffer.
+ */
+#ifdef CONFIG_PROFILER
+static inline void profiler_log_encode_int32(struct log_event_buf *buf, const int32_t data)
```

Comment:
Consider defining these functions in `profiler_nordic.c` (the functions could be defined in the same way as now). Using functions that are defined for unsigned data is actually implementation-specific for our profiler.